src/main/java/edu/cornell/kfs/module/ar/service/impl/CuContractsGrantsInvoiceCreateDocumentServiceImpl.java

```diff
@@ -111,5 +118,52 @@ protected String findContractControlAccountNumber(List<InvoiceAccountDetail> det
     protected String findTitleFormatString() {
         return getConfigurationService().getPropertyValueAsString(CuArParameterConstants.CONTRACTS_GRANTS_INVOICE_DOCUMENT_TITLE_FORMAT);
     }
+    
+    /**
+     * Added evaluation of account sub-fund not being an expenditure to existing base code evaluation
+     * of billing frequency and invoice document status when determining valid award accounts.
+     * Expenditure sub-funds associated to accounts to exclude are defined in CG system parameter
+     * CG_INVOICING_EXCLUDE_EXPENSES_SUB_FUNDS.
+     */
+    @Override
+    protected List<ContractsAndGrantsBillingAwardAccount> getValidAwardAccounts(
+            List<ContractsAndGrantsBillingAwardAccount> awardAccounts, ContractsAndGrantsBillingAward award) {
+        if (!ArConstants.BillingFrequencyValues.isMilestone(award) && !ArConstants.BillingFrequencyValues.isPredeterminedBilling(award)) {
+            List<ContractsAndGrantsBillingAwardAccount> validAwardAccounts = new ArrayList<>();
+            Set<Account> invalidAccounts = harvestAccountsFromContractsGrantsInvoices(getInProgressInvoicesForAward(award));
+
+            for (ContractsAndGrantsBillingAwardAccount awardAccount : awardAccounts) {
+                if (!invalidAccounts.contains(awardAccount.getAccount())) {
+                    if (verifyBillingFrequencyService.validateBillingFrequency(award, awardAccount)
+                        && isNotExpenditureAccount(awardAccount)) {
+                            validAwardAccounts.add(awardAccount);
+                    }
+                }
+            }
+
+            return validAwardAccounts;
+        } else {
+            return awardAccounts;
+        }
+    }
+    
+    protected boolean isNotExpenditureAccount(ContractsAndGrantsBillingAwardAccount billingAwardAccount) {
+        Account accountLinkedToAward = getAccountService().getByPrimaryId(billingAwardAccount.getChartOfAccountsCode(), billingAwardAccount.getAccountNumber());
+        if (ObjectUtils.isNotNull(accountLinkedToAward)) {
+            return !isExpenditureSubFund(accountLinkedToAward.getSubFundGroupCode());
+        }
+        return true;
+    }
+    
+    protected boolean isExpenditureSubFund(String subFundGroupCode) {
+        if(StringUtils.isNotBlank(subFundGroupCode)) {
+            Collection<String> acceptedValuesForExpenditureSubFundCodes = parameterService.getParameterValuesAsString(KFSConstants.OptionalModuleNamespaces.CONTRACTS_AND_GRANTS, 
+                    CUKFSParameterKeyConstants.ALL_COMPONENTS, CUKFSParameterKeyConstants.ContractsGrantsParameterConstants.CG_INVOICING_EXCLUDE_EXPENSES_SUB_FUNDS);
+            if(CollectionUtils.isNotEmpty(acceptedValuesForExpenditureSubFundCodes)) {
+                return acceptedValuesForExpenditureSubFundCodes.stream().anyMatch(subFundGroupCode::contains);
```

Comment:
This doesn't match the logic from your previous version of the code. If you need the case-insensitive comparisons that your old code had, I would suggest using the "subFundGroupCode::equalsIgnoreCase" method reference instead. If you just want to do a case-sensitive search on the sub-fund-group-code collection, you could just use the collection's "contains" method so that you don't need to use a stream.

Reply:
Adjusting. Thanks for catching this.